app/controllers/ticket_controller.rb

```diff
@@ -2,8 +2,12 @@ class TicketController < ApplicationController
   before_action :set_ticket, only: []
 
   def index     
-    @tickets = Ticket.where(ticket_desk_id: params[:ticket_desk_id]).where(cinema_hall_id: params[:cinema_hall_id])    
-    render json: @tickets
+    if !params[:password].blank?
+      @tickets = Ticket.where(password: params[:password]).where(ticket_desk_id: params[:ticket_desk_id]).where(cinema_hall_id: params[:cinema_hall_id])    
```

Comment:
This seems to be too long - maybe it would be better to put it in a scope in the model?

Reply:
It's not gonna be reusable, It may look like this, Im fine with it. Gonna think about it later.

Reply:
you can put multiple clauses into single `where` like `where(password: params[:password], ticket_desk_id: params[:tikcet_desk_id], cinema_hall_id: params[:cinema_hall_id])`